matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/extensions/MetricsExtensions.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022 The Matrix.org Foundation C.I.C.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.matrix.android.sdk.api.extensions
+
+import org.matrix.android.sdk.api.metrics.MetricPlugin
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.InvocationKind
+import kotlin.contracts.contract
+
+/**
+ * Executes the given [block] while measuring the transaction.
+ */
+@OptIn(ExperimentalContracts::class)
+inline fun <T> measureMetric(metricMeasurementPlugins: List<MetricPlugin>, block: () -> T): T {
+    contract {
```

Comment:
what's contract? Do we really need it (i can see it's experimental)

Reply:
It is basically a way to tell the compiler about how the function behaves. Sometimes the compiler doesn't know or is not fully aware of certain conditions or returns. So you can specify an effect when a function is invoked to help the compiler. 
In our case, it allows us to write this code: 

```kotlin
val response: KeysQueryResponse 

measureMetric(plugin) {
    response = block() // this won't be possible without contracts.
}
```